protocol-units/mempool/util/src/lib.rs

```diff
@@ -57,6 +59,11 @@ pub trait MempoolTransactionOperations {
 		Ok(mempool_transactions)
 	}
 
+	fn gc_mempool_transactions(
+		&self,
```

Comment:
I don't think this is a product of the refactor here. But, the `Ord` implementation is incorrect. The slot seconds comparison is not useful. It should be compared using the floored timestamp. 

This is also not corrected by `Transaction` type in the parent: https://github.com/movementlabsxyz/movement/blob/4947b75ce0f5b2f2927745423ebe8291fd1c41e5/util/movement-types/src/transaction.rs#L76

I think there is a gap in our unit tests that should have caught this. 

I will put in a PR to fix shortly.

Reply:
`slot_seconds` is the floored timestamp. But the field is not very useful and with the chosen slot time, it's only a factor of two of the timestamp because the timestamp in `MempoolTransaction` is rounded up to seconds.

Reply:
No, this got garbled over the refactors. You can see we're using the `slot_now` constructor which under the hood is just passing the `Self::SLOT_SECONDS` constant: https://github.com/movementlabsxyz/movement/blob/551687ea77596ad6ada55b59669effcfcb146cea/protocol-units/mempool/util/src/lib.rs#L70

Reply:
But, this is fixed in #631